src/main/java/mate/academy/service/UserService.java

```diff
@@ -14,15 +15,15 @@
      * @return - user if his email is equal to passed email.
      * Return <code>null</code> if there is no suitable user
      */
-    public static User findByEmail(String email) {
-        User l = null;
-        if (email.matches("[A-Za-z].*?@i\\.ua")) {
-            for (User a : users) {
-                if (a.getEmail() == email) {
-                    l = a;
+    public User findByEmail(String email) {
+        User user = null;
+        if (email.matches(eparametr)) {
+            for (User users : users) {
```

Comment:
лучше возвращай юзера как только его нашел, иначе, если таких юзеров будет миллион, а с нужным email будет первый, то ты сделаешь много лишних итераций:
```suggestion        
        if (email.matches(EMAIL_MATCHER))) {
            for (User user : users) {
             if (user.getEmail() == email) {
                    return user;
                }
             }
         }
```